drizzlepac/hlautils/svm_quality_analysis.py

```diff
@@ -619,6 +622,176 @@ def compare_photometry(drizzle_list, log_level=logutil.logging.NOTSET):
+        detector = total_product.detector
+        filter_name = total_product.edp_list[0].filters
+        ipppss = total_product.edp_list[0].exposure_name[0:6]
+        exposure_name = total_product.edp_list[0].exposure_name
+        exposure_filename = total_product.edp_list[0].info
+
+        # Set up the diagnostic object
+        diagnostic_obj = du.HapDiagnostic()
+        diagnostic_obj.instantiate_from_hap_obj(total_product,
+                                                data_source="{}.report_wcs".format(__taskname__),
+                                                description="WCS information")
+
+        summary_dict = {'instrument': instrument, 'detector': detector,
+                        'exposure_name': exposure_name, 'filter_name': filter_name,
+                        'exposure_filename': exposure_filename}
+        diagnostic_obj.add_data_item(summary_dict, 'General Information')
+
+        # Construct the output JSON filename
+        json_filename = '_'.join([ipppss, detector, 'svm_wcs.json'])
+
+        # Loop over all the individual exposures in the list which comprise the total product
+        for edp_object in total_product.edp_list:
+            edp_filter = edp_object.filters
+
+            # For exposures with multiple science extensions (multiple chips),
+            # generate a combined WCS
+            num_sci_ext, extname = util.count_sci_extensions(edp_object.full_filename)
+            extname_list = []
+            for x in range(num_sci_ext):
+                extname_list.append((extname, x+1))
+
+            metawcs = wcs_functions.make_mosaic_wcs(edp_object.full_filename)
+
+            # Get information from the active WCS
+            active_wcs_dict = {'primary_wcsname': metawcs.wcs.name, 'wcs_info': {'crpix1': metawcs.wcs.crpix[0],
+                               'crpix2': metawcs.wcs.crpix[1],
+                               'crval1': metawcs.wcs.crval[0], 'crval2': metawcs.wcs.crval[1],
+                               'scale': metawcs.pscale, 'orientation': metawcs.orientat}}
+
+            diagnostic_obj.add_data_item(active_wcs_dict, 'Primary WCS Information')
+
+            # Determine the possible alternate WCS solutions in the header
+            dict_of_wcskeys_names = wcsutil.altwcs.wcsnames(edp_object.full_filename, ext=1)
+
+            # Ignore the OPUS ("O") WCS, as well as the duplicate of the active WCS
+            dict_of_wcskeys_names.pop('O')
+
+            reverse_dict = {}
+            for key, value in dict_of_wcskeys_names.items():
+                reverse_dict.setdefault(value, set()).add(key)
+            keys_with_dups = set(chain.from_iterable(values for key, values in reverse_dict.items() if len(values) > 1))
+
+            list_keys = list(keys_with_dups)
+            for key in list_keys:
+                if key is ' ':
+                    continue
+                else:
+                    popkey = key
+
+            dict_of_wcskeys_names.pop(popkey)
```

Comment:
I tried running your code using the total list from ACS 10265_01. It crashed with the following traceback: 
`Traceback (most recent call last):
  File "/Users/dulude/Documents/Code/HLATransition/drizzlepac/drizzlepac/hlautils/svm_quality_analysis.py", line 857, in <module>
    report_wcs(total_obj_list)
  File "/Users/dulude/Documents/Code/HLATransition/drizzlepac/drizzlepac/hlautils/svm_quality_analysis.py", line 708, in report_wcs
    dict_of_wcskeys_names.pop(popkey)
UnboundLocalError: local variable 'popkey' referenced before assignment`

Reply:
Fixed this issue!